Makefile

```diff
@@ -1,18 +1,36 @@
-all:
+BIN_DIR ?= bin
+GO ?= go
+
+all: fmt vet check
+
+$(BIN_DIR):
+	mkdir -p $(BIN_DIR)
+
+build: $(BIN_DIR)
+	CGO_ENABLED=0 $(GO) build -ldflags='-s -w -extldflags="-static" -buildid=""' -trimpath -o $(BIN_DIR)/iptables-wrapper github.com/kubernetes-sigs/iptables-wrappers
+
+vet: ## Run go vet against code.
+	$(GO) vet ./...
+
+fmt: ## Check formatting
+	if [ "$$(gofmt -e -l . | tee /dev/tty | wc -l)" -gt 0 ]; then \
+		echo "Go files need formatting"; \
+    	exit 1; \
+	fi
 
 check: check-debian check-debian-nosanity check-debian-backports check-fedora check-alpine
 
-check-debian:
+check-debian: build
 	./test/run-test.sh --build-fail debian
 
-check-debian-nosanity:
+check-debian-nosanity: build
 	./test/run-test.sh --build-arg="INSTALL_ARGS=--no-sanity-check" --nft-fail debian-nosanity
 
-check-debian-backports:
+check-debian-backports: build
 	./test/run-test.sh --build-arg="REPO=buster-backports" debian-backports
 
-check-fedora:
+check-fedora: build
 	./test/run-test.sh fedora
 
-check-alpine:
+check-alpine: build
 	./test/run-test.sh alpine
```

Comment:
we need a `check-distroless`... that doesn't have to happen in this PR but we should make sure to file an issue about it if it doesn't

Reply:
Added tests for distroless
The dockerfile is based on github.com/kubernetes/release/blob/master/images/build/distroless-iptables, so I copied a few scripts from there to help me with installing and cleaning up dependencies.

In order to run the tests without sh (and other binaries), I ported the existing tests to Go and just compile them to a static binary.

I also added assertions around the `ip6tables` symlinks and added a couple tests for when the rules are added with `ip6tables to test the branches that detect ip6 rules.